src/main/java/com/atlassian/httpclient/apache/httpcomponents/DefaultHttpClientFactory.java

```diff
@@ -65,9 +66,18 @@ public void dispose(@Nonnull final HttpClient httpClient) throws Exception
     private HttpClient doCreate(HttpClientOptions options, ThreadLocalContextManager threadLocalContextManager)
     {
         checkNotNull(options);
-        final ApacheAsyncHttpClient httpClient = new ApacheAsyncHttpClient(eventPublisher, applicationProperties, threadLocalContextManager, options);
-        httpClients.add(httpClient);
-        return httpClient;
+        // we create only one http client instance as we don't need more
```

Comment:
what would happen if options doesn't match?

Reply:
I know but this `su...` since version 2.x (not sure which one :) ) and the introduction of the async rest jira api...
It's was a mistake to not have this common http client since this major change!
Options are really common  except the timeout which is handle with the future timeout so I do not see a major problem...